### PR TITLE
Add walletApiAdapter files to migrate to wallet-api

### PR DESCRIPTION
**Pull request by chabroA**
### 📝 Description

With the migration from live-app-sdk to wallet-api, we also need to migrate transaction adapters logic to the new wallet-api types.

cf:
- https://github.com/LedgerHQ/ledger-live/pull/1900
- https://github.com/LedgerHQ/ledger-live/pull/1901

### ❓ Context

- **Impacted projects**: `LLM`, `LLD`, `LLC`
- **Linked resource(s)**: [LIVE-5085](https://ledgerhq.atlassian.net/browse/LIVE-5085)

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-5085]: https://ledgerhq.atlassian.net/browse/LIVE-5085?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/ledger-live-common/src/families/polkadot/walletApiAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const CAN_EDIT_FEES = false;
+
+const convertToLiveTransaction = (
+  tx: WalletAPITransaction
+): Partial<Transaction> => ({
+  ...tx,
+  era: tx.era ? `${tx.era}` : undefined,
+});
+
+const getWalletAPITransactionSignFlowInfos = (
```
Comment (IAmMorrow) [inline libs/ledger-live-common/src/families/polkadot/walletApiAdapter.ts:13]: Some functions seems to always have the same prototype, would it be a good idea to define a type and import it in all those adapter modules ?

Comment (chabroA) [reply libs/ledger-live-common/src/families/polkadot/walletApiAdapter.ts:13]: Indeed the `getWalletAPITransactionSignFlowInfos` function has the same signature for all coins needing an adapter as of today. The type can indeed be extracted 👍

Comment (chabroA) [reply libs/ledger-live-common/src/families/polkadot/walletApiAdapter.ts]: cf. 6d3fea8d1dce8e07eff8b571cc9b3a017a4090b5 ✅

File: libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    amount: tx.amount,
+    recipient: tx.recipient,
+    gasPrice: tx.gasPrice,
+    userGasLimit: gasLimit,
+  };
+
+  return hasFeesProvided ? { ...liveTx, feesStrategy: "custom" } : liveTx;
+};
+
+const getWalletAPITransactionSignFlowInfos: GetWalletAPITransactionSignFlowInfos<
```
Comment (Justkant) [inline libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts:33]: No tests for this one or just an oversight ?

File: libs/ledger-live-common/src/wallet-api/types.ts [hunk 3]
```diff
@@ -62,3 +62,20 @@ export type AppManifest = {
...
+  LiveTransaction
+> = (tx: WalletAPITransaction) => {
+  canEditFees: boolean;
+  hasFeesProvided: boolean;
+  liveTx: Partial<LiveTransaction>;
+};
+
+export type AreFeesProvided<WalletAPITransaction> = (
+  tx: WalletAPITransaction
+) => boolean;
```
Comment (Justkant) [inline libs/ledger-live-common/src/wallet-api/types.ts:77]: I find it a bit misleading to use the same name for a generic and a real type, `WalletAPITransaction` is exported just above and in this lines `WalletAPITransaction` is a generic type and not the exported one.

Maybe we could simplify it, I have 2 ideas in mind:
```suggestion
export type AreFeesProvided<Transaction> = (
  tx: Transaction
) => boolean;
```
or
```suggestion
export type AreFeesProvided<T> = (
  tx: T
) => boolean;
```

And if you decide to change that we should do the same for `ConvertToLiveTransaction` and `GetWalletAPITransactionSignFlowInfos`

**Review by Justkant — CHANGES_REQUESTED** [review]
Just a small thing with the types and maybe a missing test

File: libs/ledger-live-common/src/wallet-api/types.ts (hunk 3 above)
Comment (chabroA) [reply libs/ledger-live-common/src/wallet-api/types.ts]: Fixes done here -> 0ec3e458ab6b359714579a87ef4e471258e52b45

Rename + use of `extends` 👌

File: libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts (hunk 2 above)
Comment (chabroA) [reply libs/ledger-live-common/src/families/ethereum/walletApiAdapter.ts:33]: Indeed, fixed here -> f8351c25fdf46ee25bdeabac47f9a3db85d46774 ✅

**Review by Justkant — APPROVED** [review]

**Pull request merged**